controllers/ovndbcluster_controller.go

```diff
@@ -467,35 +470,59 @@ func (r *OVNDBClusterReconciler) reconcileNormal(ctx context.Context, instance *
 
 			// Filter out headless services
 			if svc.Spec.ClusterIP != "None" {
-				// Test using hostname instead of ip for dbAddress connection
-				//serviceHostname := fmt.Sprintf("%s.%s.svc", svc.Name, svc.GetNamespace())
-				//dbAddress = append(dbAddress, fmt.Sprintf("tcp:%s:%d", serviceHostname, svc.Spec.Ports[0].Port))
-
-				internalDbAddress = append(internalDbAddress, fmt.Sprintf("tcp:%s:%d", svc.Spec.ClusterIP, svcPort))
-				raftAddress = append(raftAddress, fmt.Sprintf("tcp:%s:%d", svc.Spec.ClusterIP, svc.Spec.Ports[1].Port))
-			}
-		}
-
-		// External dbAddress if networkAttachment is used
-		if instance.Spec.NetworkAttachment != "" {
-			net := instance.Namespace + "/" + instance.Spec.NetworkAttachment
-			if netStat, ok := instance.Status.NetworkAttachments[net]; ok {
-				for _, instanceIP := range netStat {
-					dbAddress = append(dbAddress, fmt.Sprintf("tcp:%s:%d", instanceIP, svcPort))
-				}
+				internalDbAddress = append(internalDbAddress, fmt.Sprintf("tcp:%s.%s.svc.%s:%d", svc.Name, svc.Namespace, v1beta1.DNSSuffix, svcPort))
+				raftAddress = append(raftAddress, fmt.Sprintf("tcp:%s.%s.svc.%s:%d", svc.Name, svc.Namespace, v1beta1.DNSSuffix, svc.Spec.Ports[1].Port))
 			}
 		}
 
 		// Set DB Addresses
 		instance.Status.InternalDBAddress = strings.Join(internalDbAddress, ",")
-		instance.Status.DBAddress = strings.Join(dbAddress, ",")
 		// Set RaftAddress
 		instance.Status.RaftAddress = strings.Join(raftAddress, ",")
 	}
 	Log.Info("Reconciled Service successfully")
 	return ctrl.Result{}, nil
 }
 
+func getPodIPv4InNetwork(ovnPod corev1.Pod, instance *ovnv1.OVNDBCluster) (string, error) {
+	netStat, err := nad.GetNetworkStatusFromAnnotation(ovnPod.Annotations)
+	if err != nil {
+		err = fmt.Errorf("Error while getting the NAD for pod %s: %v", ovnPod.Name, err)
+		return "", err
+	}
+	var dnsIP string
+	for _, v := range netStat {
+		if v.Name == instance.Namespace+"/"+instance.Spec.NetworkAttachment {
+			// Return only IPv4 at the moment
+			for i := 0; i < len(v.IPs); i++ {
+				if !strings.Contains(v.IPs[i], ":") {
+					dnsIP = v.IPs[i]
+					break
+				}
+			}
+		}
+	}
+	return dnsIP, err
+}
+
+func deleteDNSData(ctx context.Context, helper *helper.Helper, dnsName string, namespace string) error {
+	// Delete DNS records for deleted services/pods
+	dnsData := &infranetworkv1.DNSData{}
+	err := helper.GetClient().Get(ctx, types.NamespacedName{Name: dnsName, Namespace: namespace}, dnsData)
+	if err != nil && !k8s_errors.IsNotFound(err) {
+		err = fmt.Errorf("Error while getting DNS record %s: %w", dnsName, err)
```

Comment:
Totally true, as normally two values are returned, as well as error message is long, I prefer to have two different lines to avoid long lines, but that's not the case. Done :)